src/utilities/getConfig.ts

```diff
@@ -0,0 +1,46 @@
+import { promises as fs } from 'fs';
+import * as yaml from 'js-yaml';
+import { Options } from 'semantic-release';
+
+const getConfigYaml = async (filePath: string): Promise<object> => {
+  const file = await fs.readFile(filePath, 'utf8');
+
+  const config = yaml.safeLoad(file) as string | object | undefined;
+
+  if (typeof config !== 'object') {
+    throw new Error('Invalid config file contents; not an object');
+  }
+
+  return config;
+};
+
+const getConfigJs = async (
+  filePath: string,
+  defaultOptions: Options,
+): Promise<object> => {
+  try {
+    const file = await fs.readFile(filePath, 'utf8');
```

Comment:
Would using require here be a better option? It would allow us to skip `eval`. WDYT?

Reply:
Unfortunately it doesn't work well with the compiler unfortunately. Eval shouldn't hurt though (other semantic-release plugins seem to use it), as it runs in a sandbox in CI :).